opt in scheduling on master node

kyv opened the pull request:
Please consider this pull request to allow opt in scheduling on master node ala issue #37 .

  * remove option `register-scheduable=false` from kubelet-wrapper
  * label and taint master node

Commit:
opt in scheduling on master node
  * remove option `register-scheduable=false` from kubelet-wrapper
  * label and taint master node

andrewsykim commented on 01-master.yaml:
```diff
@@ -38,7 +38,6 @@ write_files:
           --client-ca-file=/etc/kubernetes/ssl/ca.pem \
           --api-servers=http://127.0.0.1:8080 \
           --network-plugin-dir=/etc/kubernetes/cni/net.d \
-          --register-schedulable=false \
```
Let's add the labels and taints directly on the kubelet:
```
--node-labels=kubernetes.io/role=master
--register-with-taints=node-role.kubernetes.io/master=:NoSchedule
```

andrewsykim commented:
Hi @kyv thanks for the PR! I think it'll be simpler if we add the node labels and taints directly to the kubelet on master. Let me know what you think :)

kyv commented:
Great! Yes that would definitely be simpler. I was not aware of those options to kubelet.

andrewsykim commented:
Let's also add `--node-labels=kubernetes.io/role=node` to node kubelet

kyv commented:
So all nodes (master and workers) would have defined `--node-labels=kubernetes.io/role=node`?

kyv commented:
Ok [on further reading](https://github.com/kubernetes-incubator/kube-aws/issues/147) I'm seeing that people are assigning `role=master` to master and `role=node` to non-master nodes. So probably that's what you meant.

andrewsykim commented:
@kyv I meant to set `--node-labels=kubernetes.io/role=node` on the workers and `--node-labels=kubernetes.io/role=master` on the masters. `kubectl` specifically looks for the `kubernetes.io/role` label https://github.com/kubernetes/kubernetes/blob/master/pkg/printers/internalversion/printers.go#L69-L70

kyv committed:
move taints and labels to kubelet options

  * updates
https://github.com/kubernetes-digitalocean-terraform/kubernetes-digitalocean-terraform/pull/43

andrewsykim commented:
Verified this works
```
$ kubectl get no
NAME             STATUS         AGE       VERSION
10.137.112.175   Ready,master   2m        v1.7.3+coreos.0
10.137.112.187   Ready,node     2m        v1.7.3+coreos.0
10.137.240.172   Ready,node     2m        v1.7.3+coreos.0
10.137.8.25      Ready,node     2m        v1.7.3+coreos.0
$ kubectl get no 10.137.112.175 -o yaml
apiVersion: v1
kind: Node
metadata:
  annotations:
    node.alpha.kubernetes.io/ttl: "0"
    volumes.kubernetes.io/controller-managed-attach-detach: "true"
  creationTimestamp: 2017-09-03T20:10:12Z
  labels:
    beta.kubernetes.io/arch: amd64
    beta.kubernetes.io/os: linux
    kubernetes.io/hostname: 10.137.112.175
    kubernetes.io/role: master
  name: 10.137.112.175
  resourceVersion: "1046"
  selfLink: /api/v1/nodes/10.137.112.175
  uid: e4a241d5-90e3-11e7-9833-2ac69339eaa2
spec:
  externalID: 10.137.112.175
  taints:
  - effect: NoSchedule
    key: node-role.kubernetes.io/master
    timeAdded: null
```

Thanks for the PR @kyv!
